### ISSUE TITLE
Support gp/gP to leave cursor after pasted text

### ISSUE THREAD

**Issue by gerardroche** [NeoVintageous/NeoVintageous#48]
<a href="https://github.com/trkoch"><img src="https://avatars0.githubusercontent.com/u/73974?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [trkoch](https://github.com/trkoch)**
_Wednesday May 28, 2014 at 14:59 GMT_
_Originally opened as https://github.com/guillermooo/Vintageous/issues/623_

----

I'm trying to configure Vintageous such that yanking leaves me at the end of the yanked text (i.e. keep the position of the cursor while yanking). In Vim, I have the mapping

```
vmap y ygv<esc>
```

I was trying to accomplish the same in Vintageous, but it seems `gv` does not behave the same as in Vim. (Thumbs up Vintageous supports mappings!)

I wonder, using the default settings in Vintageous, how is one supposed to yank a paragraph and (repeatedly) paste it immediately after? In Vim, I can press `P` multiple times, but in Vintageous this ends in a mess.

**Comment by gerardroche** [comment]
<a href="https://github.com/trkoch"><img src="https://avatars0.githubusercontent.com/u/73974?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [trkoch](https://github.com/trkoch)**
_Wednesday May 28, 2014 at 15:02 GMT_

----

In Vim, `gv` moves to the last line of the previous visual mode selection. Vintageous, however, does not change the cursor position. I believe this is why `ygv<esc>` does not reset the cursor position to the previous position (before yanking).

**Comment by gerardroche** [comment]
<a href="https://github.com/guillermooo"><img src="https://avatars1.githubusercontent.com/u/684948?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [guillermooo](https://github.com/guillermooo)**
_Wednesday May 28, 2014 at 19:17 GMT_

----

I think `P` is placing the caret at the wrong position in Vintageous. It should leave at the begin of the paste.

**Comment by gerardroche** [comment]
<a href="https://github.com/guillermooo"><img src="https://avatars1.githubusercontent.com/u/684948?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [guillermooo](https://github.com/guillermooo)**
_Thursday May 29, 2014 at 11:43 GMT_

----

You should now be able to `P` multiple times without needing to move the caret. Does that work for your case?

**Comment by gerardroche** [comment]
<a href="https://github.com/trkoch"><img src="https://avatars0.githubusercontent.com/u/73974?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [trkoch](https://github.com/trkoch)**
_Thursday May 29, 2014 at 14:31 GMT_

----

Vintageous now mirrors Vim standard behavior, yet I still find it cumbersome to paste like this. I usually want to edit the second paragraph (e.g. when I use the previous line as a template). I believe (hope) the only thing missing is `gv` placing the cursor at the end of the selection.

**Comment by gerardroche** [comment]
<a href="https://github.com/guillermooo"><img src="https://avatars1.githubusercontent.com/u/684948?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [guillermooo](https://github.com/guillermooo)**
_Thursday May 29, 2014 at 16:47 GMT_

----

`gv` should indeed remember the selections' direction, but I don't see how that would help in your case. After  `gv`, you'd have moved away from the text just pasted.

**Comment by gerardroche** [comment]
<a href="https://github.com/trkoch"><img src="https://avatars0.githubusercontent.com/u/73974?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [trkoch](https://github.com/trkoch)**
_Thursday May 29, 2014 at 17:44 GMT_

----

Skimming through the Vim documentation, I learned about `gP`.

```
["x]gP                  Just like "P", but leave the cursor just after the new
                        text.  {not in Vi}
```

I find that pretty useful as it saves quite a lot of keystrokes.

**Comment by gerardroche** [comment]
@trkoch I've implemented `gp` and `gP` in 1.16.0 (NeoVintageous is a fork of the discontinued Vintageous plugin). There has also been lots of fixes around yanking and pasting in general. Let me know if these commands work as expected. :rocket:
